### PR TITLE
chore: Add new props for hiding past 2 years filter in date-range-picker component

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

- Add new props for hiding past 2 years filter in date-range-picker component

Fixes [#24036](https://github.com/appsmithorg/appsmith/issues/24036)

Depends on [#2286](https://github.com/appsmithorg/appsmith-ee/pull/2286)

## Type of change
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?
- Manual on storybook 

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [x] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -200,27 +202,43 @@ export function createDefaultShortcuts(
...
-    ...(useSingleDateShortcuts
+    ...(useSingleDateShortcuts || excludeShortcuts.includes("past_6_months")
       ? []
       : [createShortcut("Past 6 months", [sixMonthsAgo, today])]),
-    createShortcut(useSingleDateShortcuts ? "1 year ago" : "Past year", [
-      oneYearAgo,
-      today,
-    ]),
-    ...(useSingleDateShortcuts
+    ...(useSingleDateShortcuts || excludeShortcuts.includes("past_year")
```
Comment (albinAppsmith) [inline packages/design-system/src/DatePicker/DatePicker.tsx:233]: @tanvibhakta may be after sometime we should change this useSingleDateShortcuts flag.

**Review by albinAppsmith — APPROVED** [review]

**Comment by shadabbuchh** [comment]
Support for `today`/`yesterday` is not present in `excludeShortcuts` control.

https://github.com/appsmithorg/design-system/assets/39921438/c879694d-a10f-48b7-921d-20880d806942

**Comment by ankitakinger** [comment]
@shadabbuchh We are already having a separate prop for that one, names allowSameDay. Hence haven't included it in the exlcudeShortcuts. @albinAppsmith what do you suggest here?

**Comment by albinAppsmith** [comment]
@ankitakinger  the purpose of `sameDayFlag` is to allow user to select the same day or not. And since the new prop we added handles shortcuts, we can allow excluding shortcuts of `Today` and `Yesterday`. If it exceeds the scope of your work, you can create a task and assign it to @tanvibhakta  or me. I had already commented this for Tanvi in the PR comments.

**Review by albinAppsmith — APPROVED** [review]

**Comment by shadabbuchh** [comment]
Support for `today`/`yesterday` working now.

https://github.com/appsmithorg/design-system/assets/39921438/078fc6cb-aa25-4ddf-b78a-eb74f8341039

**Pull request merged**